OnlyM.CoreSys/GraphicsUtils.cs

```diff
@@ -237,26 +237,22 @@ public static BitmapImage ByteArrayToImage(byte[] imageData)
             return img;
         }
 
-        public static BitmapImage GetBitmapImageWithCacheOnLoad(string imageFile)
+        public static BitmapImage GetBitmapImage(string imageFile)
         {
-            var bmp = new BitmapImage();
-            
-            // BitmapCacheOption.OnLoad prevents the source image file remaining
-            // in use when the bitmap is used as an ImageSource.
-            bmp.BeginInit();
-            bmp.CreateOptions = BitmapCreateOptions.IgnoreColorProfile;
```

Comment:
We must retain this option to deal with photos that have a bad colour profile.

Reply:
No problem, I'll try to restore this handling somehow. But then, shouldn't we also take similar precautions in thumbnail generation?

Also... I doubt that systematically discarding color profile is desirable. In the past, I have personally seen a case where an image with large area of "bubble gum pink" would turn out "wine red" because ICC profiles were ignored. Granted, that was an extreme case involving CMYK to RGB conversion, but still, even in common scenario, color profiles are generally a good thing... I know however that WPF will throw when processing an image with an invalid color profile.

Would it sound good to you if I change the code to first try to load the image with its embedded color profile, then catch eventual exceptions, and only then, retry loading the image ignoring color profile? Note that I may be taking over my head now, as I am not familiar with .NET nor WPF, so maybe I will find later that doing this is is ridiculously complex, but if that sounds interesting to you, I am willing to give it a try...

Reply:
Yes, good ideas all round. Thanks for your patience.